### ISSUE TITLE
Crawler improvement

### ISSUE THREAD

**Issue by holenet** [SWPP/ttrs-backend#22]
Currently, the status of a crawler is represented as,
```json
{
    "id": 4,
    "status": "running 교양 (학문의 기초-외국어) 25/29",
    "semester": "1학기",
    "year": "2018",
    "started": "2018-05-08T22:40:37.601527+09:00"
}
```
This representation does not shows its proper status.
The first purpose of the status field is to let requester to know when the task would be finished.
So, although the crawler have several steps, it should be able to know **current progress** of entire process
or show the **estimated remaining time** to finish the task.

**Comment by lameinthebox** [comment]
Would it be enough if the crawler status shows (# pages done in current step)/(# total pages in current step) and (# pages done)/(# total pages)?

**Comment by holenet** [comment]
Yes, that will be enough.
Just printing (# pages done)/(# total pages) is simple and enough to know the current progress.

**Comment by lameinthebox** [comment]
Ok. I think I can do that.

**Comment by holenet** [comment]
In addition,
`crawler.py` seems to have some bug.
I got this error while crawling,
```json
    {
        "id": 4,
        "status": "name 'total_page' is not defined",
        "semester": "1학기",
        "started": "2018-05-08T22:40:37.601527+09:00",
        "year": "2018"
    }
```

**Comment by holenet** [comment]
Moreover,
https://github.com/SWPP/ttrs-backend/blob/a73ed0492019600811622b8257cdc9886eabd095/backend/manager/crawler.py#L99-L103
Before setting crawler's status to 'finished', it have to check its 'cancel_flag'.
If the task was cancelled, it should not set its status to 'finished'.

And, the `total_page` is not defined in this function "`run`".

**Comment by lameinthebox** [comment]
Ok. I'll go check.

**Comment by lameinthebox** [comment]
Closed with #33
